[17.0][MIG] delivery_dhl_parcel: Migration to 17.0

arantxa-s73 opened the pull request:
Standard migration delivery_dhl_parcel

Depends on:

- [ delivery_state ] https://github.com/OCA/delivery-carrier/pull/840
- [ delivery_package_number ] https://github.com/OCA/delivery-carrier/pull/839

Commit:
[14.0][ADD] delivery_dhl_parcel: DHL Parcel integration with Odoo

[UPD] Update delivery_dhl_parcel.pot

[UPD] README.rst

ioans73 committed:
[IMP] delivery_dhl_parcel: Producto DHL (B2B/B2C/R2C)

delivery_dhl_parcel 14.0.1.1.0

[UPD] Update delivery_dhl_parcel.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-delivery_dhl_parcel
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-delivery_dhl_parcel/

victoralmau committed:
[IMP] delivery_dhl_parcel: Remove dhl_parcel_last_request + dhl_parcel_last_response fields.

[UPD] Update delivery_dhl_parcel.pot

[UPD] README.rst

delivery_dhl_parcel 14.0.1.2.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-delivery_dhl_parcel
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-delivery_dhl_parcel/

Commit:
[FIX] delivery_dhl_parcel: Fix log_xml

log_xml is a carrier function, self in this context is the DhlParcelRequest class

delivery_dhl_parcel 14.0.1.2.1

ofonseca-pyming committed:
Added translation using Weblate (Catalan)

ofonseca-pyming committed:
Translated using Weblate (Catalan)

Currently translated at 2.1% (1 of 46 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-delivery_dhl_parcel
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-delivery_dhl_parcel/ca/

Commit:
[14.0][IMP] delivery_dhl_parcel: Minimum 1 kg en envíos DHL

400 Client Error: Bad Request for url: https://external.dhl.es/cimapi/api/v1/customer/shipment
El peso del envío tiene que ser como mínimo 1 kilo o como máximo 99999 kilos

delivery_dhl_parcel 14.0.1.2.2

victoralmau committed:
[IMP] delivery_dhl_parcel: Add the label format on the carrier.
TT37163

[UPD] Update delivery_dhl_parcel.pot

[UPD] README.rst

delivery_dhl_parcel 14.0.1.3.0

victoralmau committed:
[IMP] delivery_dhl_parcel: Add all customer accounts check option to end day wizard.
TT37046

[UPD] Update delivery_dhl_parcel.pot

delivery_dhl_parcel 14.0.1.4.0

victoralmau committed:
[FIX] delivery_dhl_parcel: Round the weight of the shipment to avoid errors at the end of the day.
TT37136

delivery_dhl_parcel 14.0.1.4.1

victoralmau committed:
[FIX] delivery_dhl_parcel: Save the file in the correct extension in all use cases.
TT37136

victoralmau committed:
[FIX] delivery_dhl_parcel: Set the correct label format when creating the shipment.
TT37461

delivery_dhl_parcel 14.0.1.4.2

delivery_dhl_parcel 14.0.1.4.3

Commit:
[IMP] delivery_dhl_parcel: Add maintainer

[UPD] README.rst

delivery_dhl_parcel 14.0.1.4.4

Commit:
[IMP] delivery_dhl_parcel: Cash on delivery

[UPD] Update delivery_dhl_parcel.pot

[UPD] README.rst

delivery_dhl_parcel 14.0.2.0.0

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-delivery_dhl_parcel
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-delivery_dhl_parcel/

victoralmau committed:
[IMP] delivery_dhl_parcel: Update documentation

TT40002

victoralmau committed:
[FIX] delivery_dhl_parcel: Improve changes

Create _prepare_dhl_parcel_parcel_address_info() function to reduce code.

Improve receiver + sender information:
- Remove blank spaces on partner phone
- Set phone or mobile from partner
- Concatenate street and street2 field
- Limit address field to 40 characters

TT40002

delivery_dhl_parcel 14.0.1.4.5

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (54 of 54 strings)

Translation: l10n-spain-14.0/l10n-spain-14.0-delivery_dhl_parcel
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-delivery_dhl_parcel/es/

victoralmau committed:
[MIG] delivery_dhl_parcel: Migration to 15.0

TT48078

Commit:
[UPD] Update delivery_dhl_parcel.pot

arantxa-s73 committed:
[IMP] delivery_dhl_parcel: pre-commit auto fixes

arantxa-s73 committed:
[MIG] delivery_dhl_parcel: Migration to 17.0

pedrobaeza commented:
Arantxa, evita por favor cerrar el PR cada vez que quieres hacer un cambio. Puedes simplemente empujar en la misma rama con `git push -f` para sobrescribir lo que haya. El título del PR tampoco es correcto. Cámbialo por favor.

chienandalu committed:
[FIX] delivery_dhl_parcel: show full name

If we just show the partner's name there will be cases where we'll be
losing the most important info which is the main organization where the
shipping is sent.

TT49966

hildickethan commented:
Haz cherry-pick de https://github.com/OCA/l10n-spain/pull/3659 también, el módulo a nivel  funcional es OK

ioans73 commented:
@pedrobaeza Could you add "no stale" label please?

pedrobaeza commented:
Haced rebase quitando el commit de dependencia, ya que ahora está todo fusionado.

ioans73 commented:
> Haced rebase quitando el commit de dependencia, ya que ahora está todo fusionado.

Hecho!

pedrobaeza commented on delivery_dhl_parcel/models/delivery_carrier.py:
```diff
@@ -169,6 +169,7 @@ def dhl_parcel_send_shipping(self, pickings):
                 result.append(vals)
                 continue
             vals["tracking_number"] = response.get("Tracking", "")
+            vals["response"] = response
```
Esta nueva línea para qué es? No veo ningún campo con ese nombre.

arantxa-s73 replied:
Es por este pr de la versión 14, https://github.com/OCA/l10n-spain/pull/3508

pedrobaeza replied:
Eso no se ha aprobado, y además, no se almacena en ningún sitio según veo. Quítalo por favor.

david-s73 commented:
@pedrobaeza Tengo que añadir el `response` en el campo `vals` ya que sino se pierde el numero de seguimiento del dhl así que no se rellena en ningún otro sitio ya que por eso se tiene que pasar el `reponse` al `vals`

pedrobaeza commented:
Pero insisto en que no existe ningún campo `response` declarado en ningún modelo:

![imagen](https://github.com/user-attachments/assets/a31176d4-cdd9-4647-9abe-fa4cc9695ce8)

david-s73 commented:
@pedrobaeza Así es no existe ningún campo que se llame `response` pero a lo que me refiero es que cuando la función `dhl_parcel_send_shipping` devuelve `result` necesito que este contenga el `response`. Ya que cuando hace el `result.append(vals)` si ha `vals` no se le pasas el `response` entonces se pierde.

pedrobaeza commented:
Claro, pero no se puede pasar un campo que no existe o dará error, y en ningún sitio del campo luego se quita ese valor del diccionario. Si quieres coger algún otro valor, extráelo adecuamente y lo colocas en un campo que exista.

ValentinVinagre commented:
@arantxa-s73 tenéis planteado terminar este PR?

arantxa-s73 commented:
El pr ya esta terminado. @pedrobaeza puedes revisarlo

pedrobaeza reviewed (approved):
Revisión de código

/ocabot migration delivery_dhl_parcel

@ValentinVinagre haced vosotros pruebas funcionales si podéis.

ValentinVinagre commented:
> Revisión de código
> 
> /ocabot migration delivery_dhl_parcel
> 
> @ValentinVinagre haced vosotros pruebas funcionales si podéis.

Sin problema. Lo necesitamos para un cliente, así que revisamos sin problema.

Jaimermaccione reviewed (approved):
LGTM! Tested in the Runboat environment, validating everything up to the configured setup. However, any functionality requiring valid credentials could not be tested.

pedrobaeza commented:
/ocabot merge nobump
